pkg/apis/vsphereproviderconfig/v1alpha1/vsphereclusterproviderconfig_types.go

```diff
@@ -51,6 +52,39 @@ type VsphereClusterProviderConfig struct {
 	VspherePassword         string `json:"vspherePassword,omitempty"`
 	VsphereServer           string `json:"vsphereServer"`
 	VsphereCredentialSecret string `json:"vsphereCredentialSecret,omitempty"`
+
+	// SSHAuthorizedKeys is a list of SSH public keys authorized to access
+	// deployed machines.
+	//
+	// These keys are added to the default user as determined by cloud-init
+	// in the images from which the machines are deployed.
+	//
+	// The default user for CentOS is "centos".
+	// The default user for Ubuntu is "ubuntu".
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// CAKeyPair is the key pair for ca certs.
+	CAKeyPair KeyPair `json:"caKeyPair,omitempty"`
+
+	//EtcdCAKeyPair is the key pair for etcd.
+	EtcdCAKeyPair KeyPair `json:"etcdCAKeyPair,omitempty"`
+
+	// FrontProxyCAKeyPair is the key pair for FrontProxyKeyPair.
+	FrontProxyCAKeyPair KeyPair `json:"frontProxyCAKeyPair,omitempty"`
```

Comment:
what does this do for capv?

Reply:
https://kubernetes.io/docs/setup/certificates/, ```front-proxy certificates are required only if you run kube-proxy to support an extension API server.```

Reply:
Hi @figo,

None of the included fields do anything...yet. I just wanted to get them there for others working on the refactor. We will now start to focus on connecting things to the new config fields.